src/mongo/db/repl/replica_set_config.cpp

```diff
@@ -574,7 +587,7 @@ const int ReplicaSetConfig::findMemberIndexByHostAndPort(const HostAndPort& hap)
     int x = 0;
     for (std::vector<MemberConfig>::const_iterator it = _members.begin(); it != _members.end();
          ++it) {
-        if (it->getHostAndPort() == hap) {
+        if (it->getHostInternalAndPort() == hap || it->getHostAndPort() == hap) {
```

Comment:
This method is called from the replication_coordinator and checks during heartbeat processing therefore he needs to know the **internalName**. Otherwise it will not be able to validate the receiving configuration and findSelf in the received configuration. 

Does not seem to called from any user command this is why I've changed it.